File: test/language/templates/java/templates/StructFullNameTemplateArgumentTest.java
package templates;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.IOException;

import zserio.runtime.BitSizeOfCalculator;
import zserio.runtime.ZserioError;
import zserio.runtime.io.BitStreamReader;
import zserio.runtime.io.BitStreamWriter;
import zserio.runtime.io.FileBitStreamReader;
import zserio.runtime.io.FileBitStreamWriter;

import templates.struct_full_name_template_argument.StructFullNameTemplateArgument;
import templates.struct_full_name_template_argument.Storage;
import templates.struct_full_name_template_argument.TemplatedStruct_Storage_A3A4B101;
import templates.struct_full_name_template_argument.TemplatedStruct_Storage_C76E422F;

public class StructFullNameTemplateArgumentTest
{
    @Test
    public void readWrite() throws IOException
    {
        final StructFullNameTemplateArgument structFullNameTemplateArgument =
                new StructFullNameTemplateArgument();
        structFullNameTemplateArgument.setStructExternal(new TemplatedStruct_Storage_C76E422F(
                new templates.struct_full_name_template_argument.import_storage.Storage((long)42)));
        structFullNameTemplateArgument.setStructInternal(new TemplatedStruct_Storage_A3A4B101(
                new Storage("string")));

        final BitStreamWriter writer = new FileBitStreamWriter(TEST_FILE);
        structFullNameTemplateArgument.write(writer);
        writer.close();
        final BitStreamReader reader = new FileBitStreamReader(TEST_FILE);

        final StructFullNameTemplateArgument readStructFullNameTemplateArgument =
                new StructFullNameTemplateArgument(reader);
        reader.close();
        assertTrue(structFullNameTemplateArgument.equals(readStructFullNameTemplateArgument));
    }

    private static final File TEST_FILE = new File("test.bin");
}

File: test/extensions/language/templates/cpp/StructFullNameTemplateArgumentTest.cpp
#include "gtest/gtest.h"
#include "templates/struct_full_name_template_argument/StructFullNameTemplateArgument.h"
#include "zserio/RebindAlloc.h"

namespace templates
{
namespace struct_full_name_template_argument
{

using allocator_type = StructFullNameTemplateArgument::allocator_type;
using string_type = zserio::string<allocator_type>;

TEST(StructFullNameTemplateArgumentTest, readWrite)
{
    StructFullNameTemplateArgument structFullNameTemplateArgument;
    structFullNameTemplateArgument.setStructExternal(
            TemplatedStruct_Storage_C76E422F{import_storage::Storage{42}});
    structFullNameTemplateArgument.setStructInternal(
            TemplatedStruct_Storage_A3A4B101{Storage{string_type{"string"}}});

    zserio::BitBuffer bitBuffer = zserio::BitBuffer(1024 * 8);
    zserio::BitStreamWriter writer(bitBuffer);
    structFullNameTemplateArgument.write(writer);

    zserio::BitStreamReader reader(writer.getWriteBuffer(), writer.getBitPosition(), zserio::BitsTag());
    StructFullNameTemplateArgument readStructFullNameTemplateArgument(reader);

    ASSERT_TRUE(structFullNameTemplateArgument == readStructFullNameTemplateArgument);
}

} // namespace struct_full_name_template_argument
} // namespace templates
